amazon/amazon_test.go

```diff
@@ -0,0 +1,115 @@
+package amazon_test
```

Comment:
Could we keep the test files in the same package? It helps with showing test coverage within a package and also allows for testing private functions that otherwise won't be able to get tested if it was in a different package.

Reply:
First of all, this is a well-known technique. Many Go standard packages use this technique.
https://github.com/golang/sys/blob/master/unix/mmap_unix_test.go

I think this has some advantages.
https://segment.com/blog/5-advanced-testing-techniques-in-go/

The biggest advantage is we have a chance to think if an interface is appropriate and easy-to-use.

Reply:
I think I'm willing to try this new approach after reading the article! Thanks for sharing!

Reply:
My pleasure 👍